File: 0077-combinations/0077-combinations.ts
function combine(n: number, k: number): number[][] {
    const ans: number[][] =[];
    helper(1,n,k,[],ans);
    return ans;
};
function helper(
    start:number,
    n:number,
    k:number,
    curr:number[],
    ans:number[][]
): void{
    if(curr.length==k){
        ans.push([...curr]);
        return;
    }
    for(let i:number=start;i<=n;i++){
        curr.push(i);
        helper(i+1,n,k,curr,ans);
        curr.pop();
    }
}

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna

File: 0077-combinations/0077-combinations.java
class Solution {
    public List<List<Integer>> combine(int n, int k) {
        List<List<Integer>> ans=new ArrayList<>();
        helper(1,n,k,new ArrayList<>(),ans);
        return ans;
    }
    public void helper(int start,int n,int k,List<Integer> curr,List<List<Integer>> ans){
        if(curr.size()==k){
            ans.add(new ArrayList<>(curr));
            return;
        }
        for(int i=start;i<=n;i++){
            curr.add(i);
            helper(i+1,n,k,curr,ans);
            curr.remove(curr.size()-1);
        }
    }
}

// Synced seamlessly with LeetHub Pro
// Pro features: https://bit.ly/leethubpro | Free version: https://bit.ly/leethubv4
// Get it here: https://chromewebstore.google.com/detail/bcilpkkbokcopmabingnndookdogmbna